src/topology/bornology/order.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2022 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import topology.bornology.basic
+
+/-!
+# Complete lattice structure on bornologies
+
+In this file, we put a `complete_lattice` structure on `bornology α`. Following mathlib
+conventions, we have `b₁ ≤ b₂` if and only if `id : (α, b₁) → (α, b₂)` is a `locally_bounded_map`,
+i.e larger bornologies have more bounded sets.
+
+We also define the bornology generated by a family of sets `S` as the smallest bornology for which
+all elements of `S` are bounded.
+-/
+
+
+open set filter function
+
+variables {α β γ : Type*} {b₁ b₂ : bornology α} {s t u v : set α}
+
+namespace bornology
+
+lemma cobounded_injective : injective (@cobounded α) :=
+λ b₁ b₂ h, by ext; rw h
+
+instance : partial_order (bornology α) := partial_order.lift
+  (_ : bornology α → order_dual (filter α)) cobounded_injective
+
+protected lemma le_def : b₁ ≤ b₂ ↔ b₂.cobounded ≤ b₁.cobounded := iff.rfl
+
+protected lemma le_iff : b₁ ≤ b₂ ↔ (∀ s, @is_bounded _ b₁ s → @is_bounded _ b₂ s) :=
+begin
+  rw [bornology.le_def, ← @comap_id α (@cobounded _ b₂), comap_cobounded_le_iff],
+  simp
+end
+
+instance : has_Inf (bornology α) :=
+⟨λ S, ⟨Sup (@cobounded α '' S), Sup_le (λ f ⟨b, hb, hbf⟩, hbf ▸ b.le_cofinite)⟩⟩
+
+instance : has_Sup (bornology α) :=
+⟨λ S, ⟨filter.cofinite ⊓ (Inf (@cobounded α '' S)), inf_le_left⟩⟩
+
+instance : has_inf (bornology α) :=
+⟨λ b₁ b₂, ⟨@cobounded α b₁ ⊔ @cobounded α b₂, sup_le b₁.le_cofinite b₂.le_cofinite⟩⟩
+
+instance : has_sup (bornology α) :=
+⟨λ b₁ b₂, ⟨@cobounded α b₁ ⊓ @cobounded α b₂, inf_le_of_left_le b₁.le_cofinite⟩⟩
+
+instance : has_bot (bornology α) :=
+⟨bornology.cofinite⟩
+
+instance : has_top (bornology α) :=
+⟨⟨⊥, bot_le⟩⟩
+
+instance : lattice (bornology α) :=
+{ le_sup_left := λ b₁ b₂, by {rw [bornology.le_def], exact inf_le_left},
+  le_sup_right := λ b₁ b₂, by {rw [bornology.le_def], exact inf_le_right},
+  sup_le := λ b₁ b₂ b₃ h₁₃ h₂₃, by {rw [bornology.le_def] at *, exact le_inf h₁₃ h₂₃},
+  inf_le_left := λ b₁ b₂, by {rw [bornology.le_def], exact le_sup_left},
+  inf_le_right := λ b₁ b₂, by {rw [bornology.le_def], exact le_sup_right},
+  le_inf := λ b₁ b₂ b₃ h₁₃ h₂₃, by {rw [bornology.le_def] at *, exact sup_le h₁₃ h₂₃},
+  ..bornology.partial_order,
+  ..bornology.has_inf,
+  ..bornology.has_sup }
```

Comment:
```suggestion
cobounded_injective.lattice _ (λ _ _, rfl) (λ _ _, rfl)
```
Maybe this won't give you the correct order because the order on filters is weird. In this case, write `cobounded_injective' : injective (@cobounded α ∘ of_dual)` .

Reply:
Oh right it work for the lattice. It doesn't work for the `complete_lattice` instance because the cobounded filter of `⊤` is `cofinite`, not `⊥`, which is why I did everything by hand